xde/src/xde.rs

```diff
@@ -1236,50 +1462,110 @@ fn clear_xde_underlay() -> Result<NoResp, OpteError> {
         });
     }
 
-    if let Some(underlay) = token.underlay.take() {
-        // If the underlay references have leaked/spread beyond `XdeDev`s and not
-        // been cleaned up, we committed have a fatal programming error.
-        // We aren't using `Weak` references to these types either, so no strong
-        // references could be created.
-        //
-        // We know these must succeed given that the only holders of an
-        // `Arc<XdeUnderlayPort>` are `XdeState` (whose ref we have exclusively locked)
-        // and `XdeDev` (of which none remain).
-        let name = underlay.u1.name.clone();
-        let u1 = Arc::into_inner(underlay.u1).unwrap_or_else(|| {
-            panic!("underlay u1 ({name}) must have one ref during teardown",)
-        });
+    // Clear multicast forwarding table to release any references
+    token.mcast_fwd.write().clear();
```

Comment:
I don't think it is holding any references? It contains only data, rather than any smart pointer types.